include/zephyr/posix/posix_types.h

```diff
@@ -41,6 +41,11 @@ struct pthread_attr {
 	uint32_t details[2];
 };
 
+#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_ARMCLANG_STD_LIBC) \
+	|| defined(CONFIG_ARCMWDT_LIBC)
+typedef uint32_t pid_t;
```

Comment:
Could this please be a signed type? Otherwise, PR looks great to me. Thank you!

Reply:
done, changed to int32_t